AgentInterfaces/MaintenanceSchedule.cs

```diff
@@ -38,4 +47,11 @@ public class MaintenanceEvent
 
         public int DurationInSeconds { get; set; }
     }
+
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum MaintenanceReason
```

Comment:
I would call it source, I am not sure if we need an enum for it, we can just leave it as string

Reply:
We can also add it to the MaintenanceSchedule itself, we may not need the additional class. I would call it MaintenanceSource